### PR TITLE
refactor(main):improved startup code

### PR DESCRIPTION
**Pull request by cnlangzi**

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -50,33 +50,1491 @@ github.com/0chain/errors v1.0.3 h1:QQZPFxTfnMcRdt32DXbzRQIfGWmBsKoEdszKQDb0rRM=
 github.com/0chain/errors v1.0.3/go.mod h1:xymD6nVgrbgttWwkpSCfLLEJbFO6iHGQwk/yeSuYkIc=
 github.com/0chain/gosdk v1.3.0-beta.4.0.20211109103416-4d382fe5d5d4 h1:p5EYDEE4GlAsS01nVwi+pqIU6afoiYqgQHTvRm+US1Y=
 github.com/0chain/gosdk v1.3.0-beta.4.0.20211109103416-4d382fe5d5d4/go.mod h1:8X8+3afqhRRXJcpVeUKKUFvlQ9138eafactRNUMSTL4=
+github.com/Azure/azure-pipeline-go v0.2.1/go.mod h1:UGSo8XybXnIGZ3epmeBw7Jdz+HiUVpqIlpz/HKHylF4=
+github.com/Azure/azure-pipeline-go v0.2.2/go.mod h1:4rQ/NZncSvGqNkkOsNpOU1tgoNuIlp9AfUH5G1tvCHc=
+github.com/Azure/azure-storage-blob-go v0.7.0/go.mod h1:f9YQKtsG1nMisotuTPpO0tjNuEjKRYAcJU8/ydDI++4=
```
Comment (kushthedude) [inline go.sum]: Hey, I am really not sure how these are related, can you try running `go mod tidy` on local?

File: code/go/0chain.net/blobber/zcn.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	// setup wallet
+	fmt.Print("	+ connect to miners: ")
+	if err := handler.WalletRegister(); err != nil {
+		fmt.Println(err.Error() + "\n")
+		panic(err)
+	}
+	fmt.Print("	[OK]\n")
+
+	// setup blobber (add or update) on the blockchain (multiple attempts)
+	for i := 1; i <= 10; i++ {
```
Comment (kushthedude) [inline code/go/0chain.net/blobber/zcn.go]: Do you think we should make this number configurable?

File: code/go/0chain.net/blobber/node.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	"os"
+
+	"github.com/0chain/blobber/code/go/0chain.net/core/encryption"
+	"github.com/0chain/blobber/code/go/0chain.net/core/logging"
+	"github.com/0chain/blobber/code/go/0chain.net/core/node"
+	"go.uber.org/zap"
+)
+
+func setupNode() error {
+	fmt.Print("[5/10] setup blobber")
```
Comment (kushthedude) [inline code/go/0chain.net/blobber/node.go]: Can we move this steps line into main.go where all these functions are being called? It is very ambiguos to have steps in the function themselves

File: code/go/0chain.net/blobber/main.go [hunk 4]
```diff
@@ -1,491 +1,47 @@
...
-	common.HandleShutdown(server)
-	handler.HandleShutdown(common.GetRootContext())
-
-	Logger.Info("Ready to listen to the requests")
 
-	if config.Development() {
-		go startGRPCServer(*r, grpcPortString)
-	}
+	// only enabled with "// +build integration_tests"
+	initIntegrationsTests(node.Self.ID)
```
Comment (kushthedude) [inline code/go/0chain.net/blobber/main.go]: Why did you remove the grpcServer setting up from the code?

File: code/go/0chain.net/blobber/logging.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	if config.Development() {
+		logging.InitLogging("development", logDir, "0chainBlobber.log")
+	} else {
+		logging.InitLogging("production", logDir, "0chainBlobber.log")
+	}
+
+	zcncore.SetLogFile(logDir+"/0chainBlobber.log", false)
+	zcncore.SetLogLevel(3)
+	fmt.Print("		[OK]\n")
+}
```
Comment (kushthedude) [inline code/go/0chain.net/blobber/logging.go]: Do we need this? It is actually a very trivial function, what you suggest?

File: code/go/0chain.net/blobber/main.go (hunk 4 above)
Comment (cnlangzi) [reply code/go/0chain.net/blobber/main.go]: it is moved to `startHttpServer`, not removed.

File: code/go/0chain.net/blobber/node.go (hunk 3 above)
Comment (cnlangzi) [reply code/go/0chain.net/blobber/node.go]: It was on `main`. but the code seems is not clean. so I move fmt to its function. if you think it is better on main,I will get them back.

File: go.sum (hunk 1 above)
Comment (cnlangzi) [reply go.sum]: `go mod tidy` is run. but it is not cleaned. so i think it is used by some package. please ignore it.

File: code/go/0chain.net/blobber/zcn.go (hunk 2 above)
Comment (cnlangzi) [reply code/go/0chain.net/blobber/zcn.go]: It should be unnecessary to config it by some one, right? it works for a long time. please just ignore it

File: code/go/0chain.net/blobber/logging.go (hunk 5 above)
Comment (cnlangzi) [reply code/go/0chain.net/blobber/logging.go]: Yes. It makes `main` clean.
- flags
- config
- logging
- minio
- node
- server chain
- database
- filestore
- tests
- china
- server

**Review by kushthedude — APPROVED** [review]
have refactored grpc code @cnlangzi , as it was ambiguous to start grpc from http. Do let me know if it is fine to you & feel free to merge 👍🏽

**Comment by cnlangzi** [comment]
> have refactored grpc code @cnlangzi , as it was ambiguous to start grpc from http. Do let me know if it is fine to you & feel free to merge 👍🏽

it is fine to move to new staging named `grpc` once your grpc branch is merged.
